### DISCUSSION TITLE
"Select All" functionality needed in column select

### DISCUSSION THREAD

**Discussion by cccs-rc** [apache/superset#18247] [General]
Hi all, 

Having recently deployed 1.4, we've discovered that the "Select All" option (pictured below as seen in 1.3) was removed from column SelectControl in the Explore view. It appears to have been a conscious decision made during the work done for [this SelectControl refactoring PR](https://github.com/apache/superset/pull/16510/files). However, our users rely heavily on that "Select All" functionality, so we're hoping this can be brought back in some form or fashion. 

<img src="https://user-images.githubusercontent.com/62034438/152042546-27a0095a-7e9e-448b-9f12-2f8e98698ac7.png" width="300">

Without a "Select All" functionality, users are forced to manually select all the fields they wish to include in a chart. This doesn't scale well when dealing with datasets that have 100+ columns. Our users often want to view such data sets in table charts. While they may ultimately decide to pare the number of fields down to those that are most relevant to their use cases, when users are initially developing charts, it can be critical for them to be able to easily view all the fields/data available in order to identify which fields are most relevant. 

Our users rely on this functionality so heavily that we've already jammed it back into our fork, so we've seen that things can get a little weird in the UI with the new functionality _and_ a Select All option (e.g., when selected fields are moved to the top of the column list, where should the "Select All" button go?). We're hoping we can come up with a design that will allow us to keep the "Select All" functionality while maintaining good UX. 

We're happy to do the work and put in a PR if we can get some insight into what criteria the design should meet. 

Any thoughts/insight would be much appreciated! 

Thanks!

**Comment by lmaloney** [comment]
I agree.  I just installed the current docker image for SuperSet, and Select All is missing, and it's painful! :(

**Comment by d9k** [comment]
https://github.com/apache/superset/pull/16510/files/39d86c9565c852b7ee98d82b36cb83709497e121#diff-8a4994d95cc1845fc018768aca7b712dd1b0137a1c4a146f4db6faa37b113874L69

> The "Select all" option has been removed as agreed with design and all the related tests have been removed as well

Why was this feature removed?
Please consider to return it ASAP :sob:

**Comment by kasiazjc** [comment]
Hi! Thanks for the feedback. To be honest I am not sure why this change happened, but I agree that there are multiple cases in which this would be super useful. I am going to look into though and add it to the tasks.

**Reply by srinify** [reply]
cc @villebro I remember us discussing this at some point!

**Comment by cccs-RyanK** [comment]
I just created a PR to add this feature : https://github.com/apache/superset/pull/19979

**Comment by slavanorm** [comment]
hello guys! this issue touched me a lot too.
this issue also is on top of google search for solution, therefore id post my solution here.

1. instead of creating chart, go to SQL Lab.
2. create any request for data you want, for exact answer to issue, lets create select * 
from mytable.
3. click on create chart, based on current query.
4. success, you get all the columns from query inside chart.

<img width="559" alt="Screenshot 2023-04-19 at 14 17 06" src="https://user-images.githubusercontent.com/48209342/233044787-d5c7ccfe-b8cf-4785-929c-3c1be468eda6.png">

**Comment by NicolasPA** [comment]
It seems this feature is still lacking from the chart creation in v3.1.1, 2 years later. It's particularly painful when you want to add a table chart with many columns.

@cccs-RyanK PR has been closed because the feature was being worked internally, but it doesn't seem to be the case anymore.
Would you have an update on this feature @kasiazjc?

**Comment by funkfuzz** [comment]
I am also having big pain creating a table chart with many columns. I have to add each manually and takes a lot of time. What is the reason for delaying this feature so long?

**Comment by joaoxaviergridtech** [comment]
It's also a problem when a table has date columns that update, for example, to reflect the last 7 days (the chart doesn't update getting all columns, only the ones I selected manually).

**Comment by pcgeek86** [comment]
I have a Superset dataset with 31 columns, and I don't see an option to "Select All" columns. I have to add them individually .... this would be really helpful to have.

**Comment by Gunnnn** [comment]
+1
